modules/gdscript/gdscript_compiler.cpp

```diff
@@ -1072,10 +1072,8 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 				}
 
 				if (has_setter) {
-					if (!is_in_setter) {
```

Comment:
I don't think this is a proper solution. In case it is in the setter, we already have the target properly set (since in this case it won't need to call the setter, it points directly to the member). What may be missing is just setting the proper type in the target.

But looking at this it seems the `target` isn't used at all in this case (probably it's just remains from old code that was removed during refactor), so besides setting the type earlier, this `if` block could be removed and just the `else` part is actually needed.